src/main.rs

```diff
@@ -55,10 +57,17 @@ fn main() {
     );
     let mut viewport_renderer = ViewportRenderer::new(
         &mut device,
-        SWAP_CHAIN_FORMAT,
         window_size,
         &camera.projection_matrix(),
         &camera.view_matrix(),
+        ViewportRendererOptions {
+            // FIXME: Msaa X4 is the only value currently working on
+            // all devices we tried. We should query the device
+            // capabilities (but how?!) to select the proper MSAA
+            // value.
```

Comment:
I found out this bit in Apple OpenGL docs that might help:

> The best way to find out which sample modes are supported is to call the CGL function CGLDescribeRenderer with the renderer property kCGLRPSampleModes or kCGLRPSampleAlpha. You can also determine how many samples the renderer supports by calling CGLDescribeRenderer with the renderer property kCGLRPMaxSamples.

It might give you some hints while searching for it further.

Reply:
Thanks! Usually this kind of querying is available in the graphics API, but we may be out of luck for wgpu... It may not be too terrible to have a platform specific device capability detectors we can later remove once wgpu starts supporting capability querying.

Reply:
Might be worthwhile to at least create an issue in their repo.